pkg/tuned/dialog-view.jsx

```diff
@@ -0,0 +1,199 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+define([
+    "react",
+    "base1/cockpit",
+], function(React, cockpit) {
+
+"use strict";
+var _ = cockpit.gettext;
+
+/*
+ * React template for a Cockpit dialog
+ * The primary action button is disabled while its action is in progress (waiting for promise)
+ * Expected props:
+ *  - title (string)
+ *  - body (react element, top element should be of class modal-body)
+ *  - primary_clicked
+ *     Callback function that is expected to return a promise.
+ *     parameter: callback to set the progress text (will be displayed next to spinner)
+ *  - dialog_done optional, callback with success: true if primary action succeeded, false on cancel
+ *  - primary_caption optional, defaults to 'Ok'
+ *  - cancel_clicked optional
+ *     Callback called when the dialog is canceled
+ *  - cancel_caption optional, defaults to 'Cancel'
+ *  - primary_disabled optional, defaults to false
+ *  - no_backdrop optional, skip backdrop if true
+ *  - static_error optional, always show this error
+ */
+var Dialog = React.createClass({
+    getInitialState: function() {
+        return {
+            action_in_progress: false,
+            action_progress_message: '',
+            error_message: null,
+        };
+    },
+    keyUpHandler: function(e) {
+        if (e.keyCode == 27) {
+            this.cancel_click();
+            e.stopPropagation();
+        }
+    },
+    componentDidMount: function() {
+        document.body.classList.add("modal-in");
+        document.addEventListener('keyup', this.keyUpHandler.bind(this));
+    },
+    componentWillUnmount: function() {
+        document.body.classList.remove("modal-in");
+        document.removeEventListener('keyup', this.keyUpHandler.bind(this));
+    },
+    update_progress: function(msg) {
+        this.setState({ action_progress_message: msg });
+    },
+    primary_click: function(e) {
```

Comment:
I think we should tease most of this out. There's should be no concept of a primary button, rather buttons that do things. But perhaps this is work with a separate pull request. The footer should be a separate component and be replacable.